docusaurus/docs/engineering/knowledge-base/development-lifecycle.md

```diff
@@ -4,154 +4,250 @@ sidebar_position: 7
 
-Most often, code reviews are just a matter of the maintainer enforcing consistency in the codebase, or spotting bugs you might have missed (like we all do), or asking to make code clearer for someone that has never read it before. Always feel free to ask clarifying questions, or to push back if you disagree with something.
+Once the pull request has been submitted for review, the codebase owners will review the changes made and might request edits to the code. The review process usually entails multiple rounds of back and forth between the developer and the codebase owners, and the developer often adds new commits to the pull request with the requested changes.
 
-Once your pull request is merged (aka accepted in the main codebase), you can delete your branch. Github has a easy button for you to do so.
+Once the pull request is approved, the pull request is "merged" and then "closed". Once merged, the developer's code becomes part of the `master` branch in the original `upstream` codebase. The developer can now delete the "feature branch" created in step 1.
 
-#### How to Make Requested Changes
+The feedback will vary a lot.
 
-When changes are requested, you will likely see comments on the pull request you have submitted. Simply go back to step 2, write code, commit and push until all comments are fixed. Then go back to the pull request, check that everything is in order, and ask for an additional review. You should mark the comments you have addressed as done by using the "resolve convo" button.
+- fix bugs you might have missed (it always helps to get a second pair of eyes!)
+- edit for consistency with the rest of the codebase
+- discuss an unforseen issue with the code you're adding
 
-### 5. Pulling from upstream
+...
 
-When you go back to the codebase, it's likely that more changes will have been made on the original BobaBoard codebase your code is forked from. To update your git fork to reflect the changes of "upstream" (aka the original codebase) run the following command.
+There will be a bit of back and forth, and you might get comments requesting to change some of your code or asking clarifying questions. This is normal and expected: no engineer, no matter how senior, never gets comments and questions on their code. It's just part of the job, and not a reflection of your skills.
 
-```bash   showLineNumbers
-git pull upstream master
-```
+Most often, code reviews are just a matter of the maintainer enforcing consistency in the codebase, or spotting bugs you might have missed (like we all do), or asking to make code clearer for someone that has never read it before. Always feel free to ask clarifying questions, or to push back if you disagree with something.
 
-You can now go back to step 1 and make new changes all over again.
+Once your pull request is merged (aka accepted in the main codebase), you can delete your branch. Github has a easy button for you to do so.
```

Comment:
If you use the GitHub button it only deletes the remote. You still need to delete your local branch and also prune in your local clone so that it's up-to-date with origin/upstream. I don't know if we want to get that technical here, or maybe have a callout for local repo maintenance with those commands.

`git branch -D <branch name>` to delete local and `git remote prune origin` to clean up your remotes. I use them enough I have them memorized. LOL.